src/CoxeterMatrices.jl

```diff
@@ -0,0 +1,216 @@
+export coxeter_matrix_from_group_type
+
+"""
+    coxeter_matrix_from_group_type(groupType::String)
+
+Create a coxeter matrix for a given group type. Affine groups are denoted by including a '~' character
+after a letter.
+
+#Examples
+```julia-repl
+julia> coxeter_matrix_from_group_type("A4")
+4×4 Matrix{Int64}:
+ 1  3  2  2
+ 3  1  3  2
+ 2  3  1  3
+ 2  2  3  1
+```
+```julia-repl
+julia> coxeter_matrix_from_group_type("D~4")
+5×5 Matrix{Int64}:
+ 1  2  3  2  2
+ 2  1  3  2  2
+ 3  3  1  3  3
+ 2  2  3  1  2
+ 2  2  3  2  1
+
+```
+"""
+function coxeter_matrix_from_group_type(groupType::String)
+    parsedGroupType = validate_and_parse_coxeter_group_type_string(groupType)
+    return build_coxeter_matrix_from_group_type(parsedGroupType[1], parsedGroupType[2], parsedGroupType[3])
+end
+
+"""
+    build_coxeter_matrix_from_group_type(C::Char, N::Int, isAffine::Bool)
+
+Take a given group type letter (A through I), group type integer, and boolean flag if the group is affine. Return the 
+corresponding Coxeter Matrix, if it is possible to construct.
+
+Matrices are constructed based on the diagrams in Kac's Infinite Dimensional Lie Algebras, pg. 43
+"""
+function build_coxeter_matrix_from_group_type(C::Char, N::Int, isAffine::Bool)
+    if C == 'A'
+        if !isAffine
+            M = fill(2,N,N)
+            for i = 1:N
+                M[i,i] = 1
+                if i != N
+                    M[i,i+1] = 3
+                    M[i+1,i] = 3
+                end
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('A',N+1,false)
+            M[1,N+1] = 3
+            M[N+1,1] = 3
+        end
+    elseif C =='B' || C == 'C'
+        if !isAffine
+            M = build_coxeter_matrix_from_group_type('A',N,false)
+            if N>1
+                M[N-1,N] =4
+                M[N,N-1]=4
+            else
+                error("B and C types must be of size >=2")
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('B',N+1,false)
+            if C == 'B'
+                M[1,2]=2
+                M[2,1]=2
+                M[1,3]=3
+                M[3,1]=3
+            else
+                M[1,2]=4
+                M[2,1]=4
+            end
+
+        end
+    elseif C == 'D'
+        if !isAffine 
+            if N<4 
```

Comment:
If we're being permissive (which I think we should be), then type `DN` makes sense for `N ≥ 2`. `D2` is two disconnected nodes, and `D3` is isomorphic to `A3`.